channeld/channeld.c

```diff
@@ -264,9 +264,20 @@ static const u8 *psbt_to_tx_sigs_msg(const tal_t *ctx,
 				     struct channel *channel,
 				     const struct wally_psbt *psbt)
 {
-	const struct witness_stack **ws =
-		psbt_to_witness_stacks(tmpctx, psbt,
-				       channel->opener);
+	const struct witness_stack **ws;
+	enum tx_role role;
+
+	switch (channel->opener) {
+	case LOCAL:
+		role = TX_INITIATOR;
+		break;
+	case REMOTE:
+		role = TX_ACCEPTER;
+		break;
+	default:
```

Comment:
FYI, adding a `default` totally undoes the reason for using a `switch` statement here, which is to fail to compile if/when the number of enums changes.

Reply:
There is already a third alternative (`NUM_SIDE`), the purpose of this switch was to assert that we are never passed `NUM_SIDE` (or, as unlikely as it could be, a new alternative value) since it does not make sense for the 2-alternatives enum.

Reply:
Yeah, NUM_SIDES should be a #define, not within the enum.  This is the pattern we used elsewhere.

Reply:
Just checked this. `NUM_SIDES` is a #define, and has been since https://github.com/ElementsProject/lightning/commit/2618ef10c38b6437cb3b37094f0fb4c0cb569fb1